File: app/src/main/java/in/obvious/ws/writingkotlin/nullsafety/java/NullSafety.java
package in.obvious.ws.writingkotlin.nullsafety.java;

public class NullSafety {

    public static void main(String[] args) {
        String phoneNumber = loadPhoneNumber();
        callUser(phoneNumber);
    }

    static void callUser(String phoneNumber) {

    }

    static void showError() {

    }

    static String loadPhoneNumber() {
        // Getting phone number from DB
        return null;
//        return phoneNumber;
    }
}

File: app/src/main/kotlin/in/obvious/ws/writingkotlin/nullsafety/kotlin/NullSafety.kt
package `in`.obvious.ws.writingkotlin.nullsafety.kotlin

class NullSafety {
    private val phoneNumber: String? = loadPhoneNumber()

    fun main() {
        if (phoneNumber != null) {
            callUser(phoneNumber = phoneNumber)
        }
    }

    private fun callUser(phoneNumber: String) {

    }

    private fun loadPhoneNumber(): String? {
        return null
        // return phoneNumber;
    }
}
